fix: Github OAuth Rehaul (#988)
appsmithorg/appsmith-docs: website/docs/getting-started/setup/instance-configuration/authentication/github-login.md

# GitHub OAuth

Appsmith allows you to integrate with GitHub OAuth, enabling end users to sign in to the Appsmith account using their GitHub authentication credentials.

## Register OAuth App on GitHub

To enable GitHub Sign in, log in to your [GitHub Account](https://github.com) and follow the steps below:

> 1. In your profile drop-down, navigate to **Settings > Developer Settings > OAuth Apps** and click on Register an Application to create and register a new OAuth App.

Reviewer: ```suggestion
1. In your profile drop-down, navigate to **Settings > Developer Settings > OAuth Apps** and click on **Register a new application** to create and register a new OAuth App.
```

- Admin Settings
- Environment Variables

**Admin Settings:** Click **Admin Settings > Authentication > Enable**(GitHub). Add the `Client ID` and  `Client Secret`. Click the **Save & Restart** button at the bottom of the page. 

![](/img/as_github_auth_config.png)

**Environment Variables:** Navigate to the `docker.env` file located in `<PROJECT_ROOT>/stacks/configuration/`. Update the values for the following keys:

* Configure the `APPSMITH_OAUTH2_GITHUB_CLIENT_ID` and `APPSMITH_OAUTH2_GITHUB_CLIENT_SECRET` fields with the client ID and client secret generated in the step above.
* Configure `APPSMITH_SIGNUP_ALLOWED_DOMAINS` with a second-level domain name (ex: `abc.com`), to only allow users with an email address from that domain name to log in (ex: `john@abc.com`). You can provide several domain names using a comma-separated list.
    
  ```bash
  # Example configuration in docker.env file
  # ********* GitHub OAuth **********
  APPSMITH_OAUTH2_GITHUB_CLIENT_ID=YOUR_GITHUB_CLIENT_ID
  APPSMITH_OAUTH2_GITHUB_CLIENT_SECRET=YOUR_GITHUB_CLIENT_SECRET
  APPSMITH_SIGNUP_ALLOWED_DOMAINS=exampledomain.com
  # ******************************
   ```
* Restart the Appsmith instance. 

> After these steps, your Appsmith installation should now enable GitHub Login. If you are running appsmith behind an ELB / Proxy, please configure Nginx [following this guide](/help-and-support/troubleshooting-guide/deployment-errors#oauth-sign-up-not-working).

Reviewer: The ELB/Proxy piece appears to be a Troubleshooting guide

Author: Yes, it is. We kept it like this as it was a specific error.

# GitHub OAuth

> Appsmith allows you to integrate with Google OAuth 2.0, enabling end users to sign in to the Appsmith account using their Github authentication credentials.

Author: Kept this image size to show the "Save and Restart" button.

> Appsmith allows you to integrate with Google OAuth 2.0, enabling end users to sign in to the Appsmith account using their Github authentication credentials.

Reviewer: hmm, ok got it.

Appsmith allows you to integrate with Google OAuth 2.0, enabling end users to sign in to the Appsmith account using their Github authentication credentials.

![](</img/github_login.png>)
## Enable GitHub OAuth Integration

To enable GitHub Sign in, login to your [GitHub Account](https://github.com)
To enable GitHub Sign in, login to your [GitHub Account](https://github.com) and follow the steps below:

> **1. Navigate to the** [**Organizations section**](https://github.com/settings/organizations) **of your profile and the settings of the organization under which you want to create an OAuth App**

Author: The steps I have given are valid for a specific organization. This is for a global OAuth App for a profile. I am updating the instructions.

> **1. Navigate to the** [**Organizations section**](https://github.com/settings/organizations) **of your profile and the settings of the organization under which you want to create an OAuth App**

Author: I don't know the reason. It was previously mentioned for a specific organization. I followed the same. Will confirm to from the engg. team

2. Navigate to the *Developer Settigs> OAuth Apps* in the organization settings and click on **Register an Application** button to create a new OAuth app.

![Click to expand](</img/Github_Reg_App.png>)

3. Configure the following OAuth Credentials for a web application:

    1. **Homepage URL:** the domain on which you are hosting Appsmith with HTTPS (Eg: `https://app.appsmith.com`).

    2. **Authorization Callback URL:** append `/login/oauth2/code/github` to the Homepage URL (Eg: `https://app.appsmith.com/login/oauth2/code/github`).

    ![Click to expand](</img/Github_App_Config.png>)

## Configure Appsmith instance  

:::caution
Test the Client ID and Client Secret before disabling Form login on Appsmith. This ensures that users can access Appsmith with their Github credentials. To fix an invalid Github configuration, update the environment variables.
:::
There are two ways to configure the self-hosted Appsmith instance - 
- Admin Settings
- Environment Variables

> **Admin Settings:** you can add the Client ID, Client Secret and allowed domains from [Admin Settings](/getting-started/setup/instance-configuration/). Go to *Admin Settings> Authentication> Github*, fill the required fields and click the **Save & Restart** button at the bottom of the page.

Reviewer: https://developers.google.com/style/procedures

```suggestion
**Admin Settings:** Click **Admin Settings** > **Authentication** > **Enable** (Github). Add the Client ID and  Client Secret. Click the **Save & Restart** button at the bottom of the page. 
```

![](/img/as_github_auth_config.png)

> **Environment Variables:** Navigate to the `docker.env` file located in `<PROJECT_ROOT>/stacks/configuration/`. Update the values for the following keys:

Reviewer: check how we did for Google. don't write for only Docker. 
<img width="820" alt="Screenshot 2023-03-14 at 13 34 15" src="https://user-images.githubusercontent.com/32227135/224934984-4529532e-2a09-4045-9ac2-bbe74759d90d.png">

Author: We need to tell the users where to find the config file. Just saying update the configuration file feels like half of the information. How do we solve this situation?

Reviewer: Kubernetes, AWS,  Docker have different configuration files. I don't know if we will be able to address that here. But you can say something like 

Update the values for the following keys in the instance configuration file, for example, in the `docker.env` file for Docker installation and in the `values.yaml` file for Kubernetes.

Reviewer: Hey did you add this? Can't see it in the PR

Author: Added it in google oauth. The changes in GitHub oauth file didn't get committed for some reason. Added it now.

> GitHub allows the integration of third-party applications using your GitHub Identity. Follow the below guide to configure GitHub OAuth.

Reviewer 2: Here we should highlight that Appsmith provides integration with GitHub.

> ## Configure the Github account

Reviewer 2: I think we can reword the heading to `Enable GitHub OAuth Integration`.
